### PR TITLE
Feature/my playlists component combine develop temp

### PR DESCRIPTION
**Pull request by Khrapal-Max**
Added
1. Routing MyPlaylist
- /All
- /Artist
- /Albums
2. Combine Main-HomeComponent
3. Added layout MyPlaylist
4. Binding button LikedSongs(MyPlaylist) with SongsComponent
5. Added cards:
- album-card
- playlist-card
- song-card
6. Refactoring layout Main-HomeComponent
7. Added function scrolling for rows Main-HomeComponent

### CODE REVIEW

**Review by proskurnicha — APPROVED** [review]

File: frontend/src/app/modules/main/cards/playlist-card/playlist-card.component.sass [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        text-align: left
+        color: $text
+        display: -webkit-box
+        -webkit-line-clamp: 02
+        -webkit-box-orient: vertical
+<<<<<<< HEAD:frontend/src/app/modules/main/cards/playlist-card/playlist-card.component.sass
+        overflow: hidden
+=======
+        overflow: hidden
+>>>>>>> merge:frontend/src/app/modules/main/cards/square-info-card/square-info-card.component.sass
```
Comment (NikitaKrasnovSW) [inline frontend/src/app/modules/main/cards/playlist-card/playlist-card.component.sass]: Delete merge conflicts

File: frontend/src/app/modules/main/cards/song-card/song-card.component.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+@Component({
+  selector: 'app-song-card',
+  templateUrl: './song-card.component.html',
+  styleUrls: ['./song-card.component.sass']
+})
+export class SongCardComponent {
+  @Input() iconURL! : string;
+  @Input() name! : string;
+  @Input() artist! : string;
+  @Input() podcast! : string;
```
Comment (NikitaKrasnovSW) [inline frontend/src/app/modules/main/cards/song-card/song-card.component.ts]: Fix formatting

File: frontend/src/app/modules/main/main-home/main-home.component.sass [hunk 3]
```diff
@@ -241,11 +241,15 @@ main
...
             align-items: center
             transition: background 0.4s ease
             font-size: 1.2rem
             &:hover
                 cursor: pointer
                 background: lighten($drk-rain, 8%)
+
+
+
+
```
Comment (NikitaKrasnovSW) [inline frontend/src/app/modules/main/main-home/main-home.component.sass]: Delete empty lines

File: frontend/src/app/modules/main/main-home/main-home.component.ts [hunk 4]
```diff
@@ -130,4 +130,24 @@ export class MainHomeComponent implements OnInit, OnDestroy {
     const element = document.getElementById(id);
     element?.scrollBy({ left: this._scrollingSize, behavior: 'smooth' });
   };
+
+  scrollRight = (id: string) => {
+    const element = document.getElementById(id);
+    let scrollingSize: number = 0;
+    if (id === 'album' || id === 'repeat' || id === 'playlist') {
+      const move = 255 * 6;
```
Comment (NikitaKrasnovSW) [inline frontend/src/app/modules/main/main-home/main-home.component.ts]: Move to constant

File: frontend/src/app/modules/main/main-home/main-home.component.ts [hunk 5]
```diff
@@ -130,4 +130,24 @@ export class MainHomeComponent implements OnInit, OnDestroy {
...
+      scrollingSize = move;
+    }
+    element?.scrollBy({ left: scrollingSize, behavior: 'smooth' });
+  };
+
+  scrollLeft = (id: string) => {
+    const element = document.getElementById(id);
+    let scrollingSize: number = 0;
+    if (id === 'album' || id === 'repeat' || id === 'playlist') {
+      const move = 255 * 6;
```
Comment (NikitaKrasnovSW) [inline frontend/src/app/modules/main/main-home/main-home.component.ts]: Use constant

File: frontend/src/app/modules/main/playlist/all/all.component.html [hunk 6]
```diff
@@ -1,3 +1,34 @@
...
+          [name]="playlist.name"
+          [description]="playlist.description"
+          [iconUrl]="playlist.iconURL"
+        >
+        </app-square-info-card>
+      </div>
+    </div>
+  </div>
+</div> -->
+
```
Comment (NikitaKrasnovSW) [inline frontend/src/app/modules/main/playlist/all/all.component.html]: Delete commented code

File: frontend/src/app/modules/main/search/search.component.html [hunk 7]
```diff
@@ -1 +1,3 @@
-<p>search works!</p>
+<div class="ui segment search">
+    <p>search work!</p>
+  </div>
```
Comment (NikitaKrasnovSW) [inline frontend/src/app/modules/main/search/search.component.html]: Fix formatting

File: frontend/src/app/services/playlists/playlist.service.ts [hunk 8]
```diff
@@ -10,6 +10,7 @@ import { HttpInternalService } from '../http-internal.service';
   providedIn: 'root'
 })
 export class PlaylistsService {
+  [x: string]: any;
```
Comment (NikitaKrasnovSW) [inline frontend/src/app/services/playlists/playlist.service.ts]: What is this?

**Review by NikitaKrasnovSW — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/main/cards/playlist-card/playlist-card.component.sass (hunk 1 above)
Comment (Khrapal-Max) [reply frontend/src/app/modules/main/cards/playlist-card/playlist-card.component.sass]: Done

File: frontend/src/app/modules/main/cards/song-card/song-card.component.ts (hunk 2 above)
Comment (Khrapal-Max) [reply frontend/src/app/modules/main/cards/song-card/song-card.component.ts]: Done

File: frontend/src/app/modules/main/main-home/main-home.component.sass (hunk 3 above)
Comment (Khrapal-Max) [reply frontend/src/app/modules/main/main-home/main-home.component.sass]: Done

File: frontend/src/app/modules/main/main-home/main-home.component.ts (hunk 5 above)
Comment (Khrapal-Max) [reply frontend/src/app/modules/main/main-home/main-home.component.ts]: Done

File: frontend/src/app/modules/main/playlist/all/all.component.html (hunk 6 above)
Comment (Khrapal-Max) [reply frontend/src/app/modules/main/playlist/all/all.component.html]: Done

File: frontend/src/app/modules/main/search/search.component.html (hunk 7 above)
Comment (Khrapal-Max) [reply frontend/src/app/modules/main/search/search.component.html]: Done

File: frontend/src/app/services/playlists/playlist.service.ts (hunk 8 above)
Comment (Khrapal-Max) [reply frontend/src/app/services/playlists/playlist.service.ts]: I dont know. Removed.

File: frontend/src/app/modules/main/main-home/main-home.component.ts (hunk 4 above)
Comment (Khrapal-Max) [reply frontend/src/app/modules/main/main-home/main-home.component.ts]: Done

File: frontend/src/app/modules/main/cards/song-card/song-card.component.ts [hunk 9]
```diff
@@ -6,8 +6,12 @@ import { Component, Input } from '@angular/core';
...
-  @Input() artist! : string;
-  @Input() podcast! : string;
+  @Input()
+  iconURL : string;
+  @Input()
+  name : string;
+  @Input()
+  artist : string;
+  @Input()
+  podcast : string;
```
Comment (NikitaKrasnovSW) [inline frontend/src/app/modules/main/cards/song-card/song-card.component.ts]: Fix formatting

**Review by NikitaKrasnovSW — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/main/cards/song-card/song-card.component.ts (hunk 9 above)
Comment (Khrapal-Max) [reply frontend/src/app/modules/main/cards/song-card/song-card.component.ts]: Done

**Review by NikitaKrasnovSW — APPROVED** [review]

**Pull request merged**